src/js/components/TaskView.js

```diff
@@ -60,6 +61,24 @@ class TaskView extends mixin(SaveStateMixin, StoreMixin) {
     super.componentWillMount();
   }
 
+  componentWillReceiveProps(nextProps) {
+    let {filterByStatus, searchString} = this.state;
+    let tasks = this.getFilteredTasks(filterByStatus, nextProps.tasks, searchString);
+    let prevTasks = this.getFilteredTasks(filterByStatus, this.props.tasks, searchString);
+
+    if (!deepEqual(tasks, prevTasks)) {
```

Comment:
I feel that the two calls to `getFilteredTasks` and the `deepEqual` is going to get pretty expensive with thousands of tasks.
@daiyiwhale would you mind doing some performance tests for this method, with 1k, 10k and 100k tasks?